src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppSelector = useSelector.withTypes<RootState>();
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
Similarly, `useDispatch.withTypes<AppDispatch>()` is incorrect. The `useDispatch` hook does not have a `withTypes` method. You should use `useDispatch` directly, and TypeScript will infer the types based on the `AppDispatch` type.